polkadot/node/network/statement-distribution/src/v2/mod.rs

```diff
@@ -826,7 +826,21 @@ pub(crate) fn handle_deactivate_leaves(state: &mut State, leaves: &[Hash]) {
 	// clean up sessions based on everything remaining.
 	let sessions: HashSet<_> = state.per_relay_parent.values().map(|r| r.session).collect();
 	state.per_session.retain(|s, _| sessions.contains(s));
-	state.unused_topologies.retain(|s, _| sessions.contains(s));
+
+	let last_session_index = state
+		.unused_topologies
+		.keys()
+		.max()
+		.map(|session_index| *session_index)
+		.unwrap_or(0);
```

Comment:
This works but feels clunky, we could just rewrite the retain below like this: `last_session_index == Some(*s)`

Reply:
Done.